editor/src/document/document_message_handler.rs

```diff
@@ -669,30 +670,36 @@ impl DocumentMessageHandler {
 		}]);
 
 		let document_mode_layout = WidgetLayout::new(vec![LayoutRow::Row {
-			widgets: vec![WidgetHolder::new(Widget::DropdownInput(DropdownInput {
-				entries: vec![vec![
-					DropdownEntryData {
-						label: DocumentMode::DesignMode.to_string(),
-						icon: DocumentMode::DesignMode.icon_name(),
-						..DropdownEntryData::default()
-					},
-					DropdownEntryData {
-						label: DocumentMode::SelectMode.to_string(),
-						icon: DocumentMode::SelectMode.icon_name(),
-						on_update: WidgetCallback::new(|_| DialogMessage::RequestComingSoonDialog { issue: Some(330) }.into()),
-						..DropdownEntryData::default()
-					},
-					DropdownEntryData {
-						label: DocumentMode::GuideMode.to_string(),
-						icon: DocumentMode::GuideMode.icon_name(),
-						on_update: WidgetCallback::new(|_| DialogMessage::RequestComingSoonDialog { issue: Some(331) }.into()),
-						..DropdownEntryData::default()
-					},
-				]],
-				selected_index: Some(self.document_mode as u32),
-				draw_icon: true,
-				..Default::default()
-			}))],
+			widgets: vec![
+				WidgetHolder::new(Widget::DropdownInput(DropdownInput {
+					entries: vec![vec![
+						DropdownEntryData {
+							label: DocumentMode::DesignMode.to_string(),
+							icon: DocumentMode::DesignMode.icon_name(),
+							..DropdownEntryData::default()
+						},
+						DropdownEntryData {
+							label: DocumentMode::SelectMode.to_string(),
+							icon: DocumentMode::SelectMode.icon_name(),
+							on_update: WidgetCallback::new(|_| DialogMessage::RequestComingSoonDialog { issue: Some(330) }.into()),
+							..DropdownEntryData::default()
+						},
+						DropdownEntryData {
+							label: DocumentMode::GuideMode.to_string(),
+							icon: DocumentMode::GuideMode.icon_name(),
+							on_update: WidgetCallback::new(|_| DialogMessage::RequestComingSoonDialog { issue: Some(331) }.into()),
+							..DropdownEntryData::default()
+						},
+					]],
+					selected_index: Some(self.document_mode as u32),
+					draw_icon: true,
+					..Default::default()
+				})),
+				WidgetHolder::new(Widget::Separator(Separator {
+					separator_type: SeparatorType::Section,
+					direction: SeparatorDirection::Horizontal,
+				})),
```

Comment:
Not usually, but this is a case where we needed to have the separator and we could either define it in Vue or Rust. If we define it in Vue, it shows up before the layout is sent to the frontend and there's a weird separator just sitting there on its own for a second. I wouldn't say standardizing is really the relevant concept here because it's reasonably rare that we will have two different layouts defined side-by-side (usually they'll be on opposite sides of the row, or in different rows).